src/mnemonic-key.ts

```diff
@@ -16,6 +17,25 @@ export class MnemonicKey extends Key {
       mnemonic = Wallet.generateMnemonic()
     }
 
-    this.addByMnemonic(mnemonic, options.index || 0)
+    this._addByMnemonic(mnemonic, options)
+  }
+
+  private _addByMnemonic(
+    phrase: string,
+    { index = 0, derivationPath = HDPath, numberOfAddresses = 1 }: MnemonicOptions,
+  ) {
+    if (!HDNode.isValidMnemonic(phrase)) {
+      throw new Error(`Invalid mnemonic phrase: ${phrase}`)
+    }
+
+    const seed = bip39.mnemonicToSeed(phrase)
+    const hdKey = hdkey.fromMasterSeed(seed)
+
+    for (let i = 0; i < numberOfAddresses; i++) {
+      const childKey =
+        index !== 0 ? hdKey.derive(`${derivationPath}${index + i}`) : hdKey.derive(`${derivationPath}${index}`)
+      const address = `0x${childKey.privateKey.toString('hex')}`
```

Comment:
why are we calling this `address`? I think this should be `privateKey` and we should remove the `0x`, since that's for accounts.

Reply:
You're right, I'll change it right away!